File: Backend/Java env/auth-service/Eclothing/src/main/java/com/example/demo/controller/UserController.java
package com.example.demo.controller;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.example.demo.dto.SellerProfileResponse;
import com.example.demo.dto.UpdateCustomerRequest;
import com.example.demo.dto.UpdateSellerRequest;
import com.example.demo.entities.User;
import com.example.demo.service.UserService;

@RestController
@RequestMapping("/users")
public class UserController {
	 	@Autowired
	    private UserService userService;

	 	// Get All Users
	    @GetMapping
	    public List<User> getAllUsers() {
	        return userService.getAllUsers();
	    }
	    
	    // Get User By Id
	    @GetMapping("/{id}")
	    public User getUserById(@PathVariable Integer id) {
	        return userService.getUserById(id);
	    }
	    
	    @DeleteMapping("/{id}")
	    public String deleteUser(@PathVariable Integer id) {
	        return userService.deleteUser(id);
	    }
	    
	    @PutMapping("/{id}")
	    public String updateCustomer(@PathVariable Integer id, @RequestBody UpdateCustomerRequest request) {
	        return userService.updateCustomer(id, request);
	    }
	    
	    @PutMapping("/seller/{id}")
	    public String updateSeller(@PathVariable Integer id,@RequestBody UpdateSellerRequest request) {
	        return userService.updateSeller(id, request);
	    }
	    
	    @GetMapping("/seller/profile/{id}")
	    public SellerProfileResponse getSellerProfile(@PathVariable Integer id){
	        return userService.getSellerProfile(id);
	    }
}

File: Backend/DotNet env/AuthService/Controllers/UserController.cs
using System.Threading.Tasks;
using AuthService.Models.DTOs;
using AuthService.Services;
using Microsoft.AspNetCore.Mvc;

namespace AuthService.Controllers
{
    [ApiController]
    [Route("users")]
    public class UserController : ControllerBase
    {
        private readonly IUserService _userService;

        public UserController(IUserService userService)
        {
            _userService = userService;
        }

        // Get All Users
        [HttpGet]
        public async Task<IActionResult> GetAllUsers()
        {
            var users = await _userService.GetAllUsersAsync();
            return Ok(users);
        }

        // Get User By Id
        [HttpGet("{id:int}")]
        public async Task<IActionResult> GetUserById(int id)
        {
            var user = await _userService.GetUserByIdAsync(id);
            if (user == null)
            {
                return NotFound(new { message = "User Not Found" });
            }
            return Ok(user);
        }

        // Soft Delete User
        [HttpDelete("{id:int}")]
        public async Task<IActionResult> DeleteUser(int id)
        {
            var result = await _userService.DeleteUserAsync(id);
            if (result == "User Not Found")
            {
                return NotFound(new { message = result });
            }
            return Ok(result);
        }

        // Update Customer
        [HttpPut("{id:int}")]
        public async Task<IActionResult> UpdateCustomer(int id, [FromBody] UpdateCustomerRequest request)
        {
            var result = await _userService.UpdateCustomerAsync(id, request);
            if (result.Contains("Not Found"))
            {
                return NotFound(new { message = result });
            }
            return Ok(result);
        }

        // Update Seller
        [HttpPut("seller/{id:int}")]
        public async Task<IActionResult> UpdateSeller(int id, [FromBody] UpdateSellerRequest request)
        {
            var result = await _userService.UpdateSellerAsync(id, request);
            if (result.Contains("Not Found"))
            {
                return NotFound(new { message = result });
            }
            return Ok(result);
        }

        // Get Seller Profile
        [HttpGet("seller/profile/{id:int}")]
        public async Task<IActionResult> GetSellerProfile(int id)
        {
            var profile = await _userService.GetSellerProfileAsync(id);
            if (profile == null)
            {
                return NotFound(new { message = "Seller Profile Not Found" });
            }
            return Ok(profile);
        }
    }
}
